File: hw/acuña/u2/HW17TestCaseSolution/ComplexOperationsJS/tests/PersonController.test.js
const PersonController = require('../ec.edu.espe.complexoperations.controler/PersonController');

test('computeAgeInDays', () => {
    const birthDate = new Date("2001-08-07");
    const expResult = 8373; // La edad exacta puede variar ligeramente debido a diferencias en el cálculo exacto de días
    const result = PersonController.computeAgeInDays(birthDate);
    expect(result).toEqual(expResult);
});

test('computeAgeInYears', () => {
    const birthDate = new Date("2001-08-07");
    const expResult = 22;
    const result = PersonController.computeAgeInYears(birthDate);
    expect(result).toEqual(expResult);
});

File: ws/lisintuña/u2/WS23/ComplexOperations/src/test/java/ec/edu/espe/complexoperations/controller/PersonControllerTest.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/UnitTests/JUnit5TestClass.java to edit this template
 */
package ec.edu.espe.complexoperations.controller;

import java.time.LocalDate;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
 *
 * @author Eduardo Segarra, TheJavaBandits, DCCO-ESPE
 */
public class PersonControllerTest {
    
    public PersonControllerTest() {
    }

    @Test
    public void testComputeAgeInDays() {
        System.out.println("computeAgeInDays");
        LocalDate birthDate = LocalDate.parse("2004-1-31");
        int expResult = 7464;
        int result = PersonController.computeAgeInDays(birthDate);
        assertEquals(expResult, result);
    }

    @Test
    public void testComputeAgeInYears() {
        System.out.println("computeAgeInYears");
        LocalDate birthDate = LocalDate.parse("2004-1-31");
        int expResult = 20;
        int result = PersonController.computeAgeInYears(birthDate);
        assertEquals(expResult, result);
    }
    
}
